docs/lib/datastore/fragments/ios/datastore-events.md

```diff
@@ -0,0 +1,50 @@
+## Usage
+For instance, to see if the network status is active, you could set up the following listener or Subscriber (using Combine):
+
+<amplify-block-switcher>
+
+<amplify-block name="Listener (iOS 11+)">
+
+```swift
+let hubEventListener = Amplify.Hub.listen(to: .dataStore) { event in
+    if event.eventName == HubPayload.EventName.DataStore.networkStatus {
+        guard let networkStatus = event.data as? NetworkStatusEvent else {
+            print("Failed to cast data as NetworkStatusEvent")
+            return
+        }
+        print("User receives a network connection status: \(networkStatus.active)")
+    }
+}
+```
+
+</amplify-block>
+
+<amplify-block name="Combine (iOS 13+)">
+
+```swift
+let hubEventSubscriber = Amplify.Hub.publisher(for: .dataStore).sink { event in
+    if event.eventName == HubPayload.EventName.DataStore.networkStatus {
+        guard let networkStatus = event.data as? NetworkStatusEvent else {
+            print("Failed to cast data as NetworkStatusEvent")
+            return
+        }
+        print("User receives a network connection status: \(networkStatus.active)")
+    }
+}
+```
+
+</amplify-block>
+
+</amplify-block-switcher>
+
+<amplify-callout>
+
+For `networkStatus` event, an initial value in which `active` is always set to `false` is received, but shortly after, you will receive a updated value which represents the real status of network connectivity.
```

Comment:
hhmm let's discuss this. This would represent a platform implementation difference and can be quite confusing.

Is there a mitigation we can put in our implementation?